### PR TITLE
TEST-#3205: Fix `push-to-master` failed tests

### LINKED ISSUE #3205: `push-to-master` docker tests fail
After docker image creation tests execution fails with messages:
```
modin/config/pubsub.py:230: in get
    raw = cls._get_raw_from_config()
modin/config/envvars.py:46: in _get_raw_from_config
    return os.environ[cls.varname]
anaconda3/lib/python3.7/os.py:681: in __getitem__
    raise KeyError(key) from None
E   KeyError: 'MODIN_ENGINE'

During handling of the above exception, another exception occurred:
modin/experimental/xgboost/test/test_default.py:23: in <module>
    Engine.get() == "Ray",
modin/config/pubsub.py:232: in get
    cls._value = cls._get_default()
modin/config/envvars.py:87: in _get_default
    import ray
anaconda3/lib/python3.7/site-packages/ray/__init__.py:77: in <module>
    import ray._raylet  # noqa: E402
python/ray/_raylet.pyx:108: in init ray._raylet
    ???
anaconda3/lib/python3.7/site-packages/ray/exceptions.py:7: in <module>
    from ray.core.generated.common_pb2 import RayException, Language, PYTHON
anaconda3/lib/python3.7/site-packages/ray/core/generated/common_pb2.py:22: in <module>
    create_key=_descriptor._internal_create_key,
E   AttributeError: module 'google.protobuf.descriptor' has no attribute '_internal_create_key'
```

### PR DESCRIPTION
**Pull request by amyskov**
Signed-off-by: Alexander Myskov <alexander.myskov@intel.com>

## What do these changes do?
It was found that creation of environment with all needed dependencies in the docker container with Ray nightly build is complex, takes much time (~4 hours) and strongly depends on the needed packages list and it's versions. Moreover after environment creation is finished, some packages work incorrectly and tests fail with message `AttributeError: module 'google.protobuf.descriptor' has no attribute '_internal_create_key'`. So, to avoid these problems, pip installable Ray nightly build instead of Ray container was used as it is done in `push-to-master` `test-ray-client` CI job.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] signed commit with `git commit -s` 
- [x] Resolves #3205  
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by amyskov** [comment]
Hi @modin-project/modin-core and other interested reviewers, this PR is ready for review. For now it marked as draft to prevent changes merge before we change `on` trigger from `pull_request` to `push`.

File: .github/workflows/push-to-master.yml [hunk 1]
```diff
@@ -1,39 +1,49 @@
...
-            pytest modin/pandas/test/test_general.py
-            pytest modin/pandas/test/test_io.py
-            pytest modin/experimental/pandas/test/test_io_exp.py
+          activate-environment: modin
+          environment-file: requirements/requirements-no-engine.yml
+          python-version: 3.7
+          channel-priority: strict
+          use-only-tar-bz2: true # IMPORTANT: This needs to be set for caching to work properly!
+      - name: install Ray nightly build
+        run: pip install https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-2.0.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl
```
Comment (vnlitvinov) [inline .github/workflows/push-to-master.yml:21]: What is the reason for the change from docker to `pip install`? Also this tests 2.0.0 and would have to be updated when they release it...

Comment (amyskov) [reply .github/workflows/push-to-master.yml:21]: pip installable ray version was used because of problems with creating environment inside Ray docker container (shortly described in the PR header). Yes, after new Ray release we will need to correct this line, but we already have hardcoded `pip install` command in the `test-ray-client` CI job.

**Review by vnlitvinov — DISMISSED** [review]
LGTM

**Review by vnlitvinov — APPROVED** [review]

**Review by YarShev — APPROVED** [review]
LGTM but I have a question.

> It was found that creation of environment with all needed dependencies in the docker container with Ray nightly build is complex, takes much time (~4 hours) and strongly depends on the needed packages list and it's versions.

Is it expected behavior (~4 hours)? Shouldn't we ask guys from Ray, for example, about that?

**Comment by amyskov** [comment]
> LGTM but I have a question.
> 
> > It was found that creation of environment with all needed dependencies in the docker container with Ray nightly build is complex, takes much time (~4 hours) and strongly depends on the needed packages list and it's versions.
> 
> Is it expected behavior (~4 hours)? Shouldn't we ask guys from Ray, for example, about that?

Initially environment creation time was much lower, but creation time increase was possible. The reason is that Ray docker container contain some additional packages for testing and building and updating this environment with Modin dependencies causes long and complex dependencies solving step (most probably environment creation time was increased because of version unpinning for some of Modin dependencies). Not sure Ray team can help us with this problem.

**Comment by YarShev** [comment]
@devin-petersohn , thoughts? Are you okay with the changes?

**Comment by vnlitvinov** [comment]
I think we should be fine as we're using [Ray-provided way of installing nightly packages](https://docs.ray.io/en/master/installation.html#daily-releases-nightlies). I'm merging this PR.
